--- a/chrome/browser/prefs/browser_prefs.cc
+++ b/chrome/browser/prefs/browser_prefs.cc
@@ -1012,6 +1012,12 @@ inline constexpr char kPendingMetricsRep
     "pending.cros.metrics.metricsReportingLevel";
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
+// Deprecated 04/2026, remove on or after 07/2026.
+inline constexpr char kQuillVerticalCollapsed[] =
+    "quill.browser.vertical_collapsed";
+inline constexpr char kQuillVerticalUncollapsedWidth[] =
+    "quill.browser.vertical_uncollapsed_width";
+
 // Register local state used only for migration (clearing or moving to a new
 // key).
 void RegisterLocalStatePrefsForMigration(PrefRegistrySimple* registry) {
@@ -1326,6 +1332,10 @@ void RegisterProfilePrefsForMigration(
   registry->RegisterInt64Pref(kSafeBrowsingModuleLastCooldownStartAt, 0);
   registry->RegisterBooleanPref(kSafeBrowsingModuleOpened, false);
 
+  // Quill Deprecated 04/2026.
+  registry->RegisterBooleanPref(kQuillVerticalCollapsed, false);
+  registry->RegisterIntegerPref(kQuillVerticalUncollapsedWidth, -1);
+
   // Deprecated 04/2026.
   registry->RegisterIntegerPref(kPreallocatedAddressesVersion, 1);
   registry->RegisterListPref(kPreallocatedAddresses);
@@ -2675,6 +2685,25 @@ void MigrateObsoleteProfilePrefs(PrefSer
   tabs::MigrateEverythingMenuPinnedToTabstripPref(profile_prefs);
 #endif
 
+  // Added 04/2026.
+  if (!profile_prefs->HasPrefPath(prefs::kVerticalTabsCollapsedState)) {
+    if (const base::Value* user_value =
+            profile_prefs->GetUserPrefValue(kQuillVerticalCollapsed)) {
+      bool value = user_value->GetBool();
+      profile_prefs->SetBoolean(prefs::kVerticalTabsCollapsedState, value);
+    }
+  }
+  profile_prefs->ClearPref(kQuillVerticalCollapsed);
+
+  if (!profile_prefs->HasPrefPath(prefs::kVerticalTabsUncollapsedWidth)) {
+    if (const base::Value* user_value =
+            profile_prefs->GetUserPrefValue(kQuillVerticalUncollapsedWidth)) {
+      int value = user_value->GetInt();
+      profile_prefs->SetInteger(prefs::kVerticalTabsUncollapsedWidth, value);
+    }
+  }
+  profile_prefs->ClearPref(kQuillVerticalUncollapsedWidth);
+
   // Please don't delete the following line. It is used by PRESUBMIT.py.
   // END_MIGRATE_OBSOLETE_PROFILE_PREFS
 
